res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2280 @@
+
+// If true, the sync button blinks with the beat. 
+// If false, the sync button is lit when sync is enabled.
+PioneerDDJSX.blinkingSync = false; //default: false
+
+// Sets the jogwheels sensivity. 1 is default, 2 is twice as sensitive, 0.5 is half as sensitive.
+PioneerDDJSX.jogwheelSensivity = 1; //default: 1
+
+// Sets how much more sensitive the jogwheels get when holding shift.
+// Set to 1 to disable jogwheel sensitivity increase when holding shift.
+PioneerDDJSX.jogwheelShiftMultiplier = 10; //default: 10
+
+// Sets the default speed slider range (0.08 = 8%).
+PioneerDDJSX.speedSliderRange = 0.08; //default: 0.08
+
+// Cuts level-meter low frequencies and expands upper frequencies. Examples:
+// 0.25 -> only signals greater 25%, expanded to full range
+// 0.5 -> only signals greater 50%, expanded to full range
+PioneerDDJSX.cutVumeter = 0; //default: 0
+
+// If true, level-Level twinkle if AutoDJ is enabled.
+PioneerDDJSX.twinkleVumeterAutodjOn = true; //default: true
+// If true, selected track will be added to AutoDJ queue-top on pressing rotary selector,
+// else track will be added to AutoDJ queue-bottom.
+PioneerDDJSX.autoDJAddTop = false; //default: false
+// Sets the duration of sleeping between AutoDJ actions if AutoDJ is enabled [ms].
+PioneerDDJSX.autoDJTickInterval = 1000; //default: 1000
+// Sets the maximum adjustment of BPM allowed for beats to sync if AutoDJ is enabled [BPM].
+PioneerDDJSX.autoDJMaxBpmAdjustment = 10; //default: 10
+// If true, AutoDJ queue is being shuffled after skipping a track.
+// When using a fixed set of tracks without manual intervention, some tracks may be unreachable, 
+// due to having an unfortunate place in the queue ordering. This solves the issue.
+PioneerDDJSX.autoDJShuffleAfterSkip = false; //default: false
+
+// If true, by releasing rotary selector, 
+// track in preview player jumps forward to "jumpPreviewPosition". 
+PioneerDDJSX.jumpPreviewEnabled = true; //default: true
+PioneerDDJSX.jumpPreviewPosition = 0.3; //default: 0.3
+
+// If true, LoadSamplerBank file dialog will show up at running init.
+PioneerDDJSX.samplerShowLoadDialog = true; //default: true
+// If true, pad press in SAMPLER-PAD-MODE repeatedly causes sampler to play 
+// loaded track from cue-point, else it causes to play loaded track from the beginning.
+PioneerDDJSX.samplerCueGotoAndPlay = false; //default: false
+
+// If true, PFL / Cue (headphone) is being activated by loading a track into certain deck.
+PioneerDDJSX.autoPFL = true; //default: true
+
+// If true, new in Mixxx 2.1 introduced library controls will be used (buggy),
+// else old playlist controls will be used.
+PioneerDDJSX.useNewLibraryControls = false; //default: false
+
+
+///////////////////////////////////////////////////////////////
+//               INIT, SHUTDOWN & GLOBAL HELPER              //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSX.init = function(id) {
+    PioneerDDJSX.syncRate = 0;
```

Comment:
I'm not sure if `this` works in the init and shutdown functions. I think `this` might be forcibly set to the global object for those functions, but I'd have to test.